### PR TITLE
Update to build under Swift 3.0 and Swift 3.1+

### PR DESCRIPTION
**Pull request by aaroncrespo**
The PR should summarize what was changed and why. Here are some questions to
help you if you're not sure:

 - What behavior was changed?

Updated CWLPreconditionTesting

 - What code was refactored / updated to support this change?

CWLPreconditionTesting associated files.

 - What issues are related to this PR? Or why was this change introduced?

Support lack of setjmp for swift 3.1. Now builds on Xcode 8.3 beta #395

Checklist - While not every PR needs it, new features should consider this list:

 - [x] Does this have tests?
 - [x] Does this have documentation?
NA
 - [x] Does this break the public API (Requires major version bump)?
NA
 - [x] Is this a new feature (Requires minor version bump)?
NA (patch)

### CODE REVIEW

File: Sources/Lib/CwlPreconditionTesting/CwlPreconditionTesting/mach_excServer.c [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 /* Module mach_exc */
 
+#if defined(__x86_64)
```
Comment (aaroncrespo) [inline Sources/Lib/CwlPreconditionTesting/CwlPreconditionTesting/mach_excServer.c]: un sure about this change, this was a dif from the existing code but is present in CwlPreconditionTesting. seems to be this is modifying a generated file but I am not well versed in these files.

**Comment by aaroncrespo** [comment]
There might still be some dust to settle on the Cwl Side.

**Comment by aaroncrespo** [comment]
I'm not 100% sure why this is failing the cocoa pods test other than the possibility of

First failure was due to tvOS tests not running but it looks like this project doesn't support tvOS.
I added files to the tvOS target and that tests passes but now cocoa pods is failing because the files required to pass the tvOS tests are not part of the umbrella header because of conditional:
```
#if !TARGET_OS_TV
    #import "mach_excServer.h"
#endif
```

**Comment by jeffh** [comment]
CWL doesn't support this, so it needs to be excluded from that target IIRC

**Comment by aaroncrespo** [comment]
Cool, all checks pass now.

**Comment by jeffh** [comment]
Thanks @aaroncrespo!

**Pull request merged**